src/test/java/org/opensearch/sdk/TestExtensionsRunner.java

```diff
@@ -11,55 +11,124 @@
+import org.opensearch.transport.Transport;
 
 public class TestExtensionsRunner extends OpenSearchTestCase {
 
     private ExtensionsRunner extensionsRunner;
     private Settings settings;
+    private TransportService transportService;
 
     @BeforeEach
-    public void setUp() throws IOException {
-
+    public void setUp() throws Exception {
         this.extensionsRunner = new ExtensionsRunner();
-        this.settings = Settings.builder().put("node.name", "MainScriptTests").build();
+        this.settings = Settings.builder().put("node.name", "sdk").build();
+        this.transportService = spy(
+            new TransportService(
+                Settings.EMPTY,
+                mock(Transport.class),
+                null,
+                TransportService.NOOP_TRANSPORT_INTERCEPTOR,
+                x -> null,
+                null,
+                Collections.emptySet()
+            )
+        );
     }
 
     // test ExtensionsRunner getTransportService return type is transport service
     @Test
-    public void testGetTransportService() throws IOException {
+    public void testGetTransportService() throws Exception {
         assert (extensionsRunner.createTransportService(settings) instanceof TransportService);
     }
 
     // test manager method invokes start on transport service
     @Test
-    public void testTransportServiceStarted() throws IOException {
-
-        // retrieve and mock transport service
-        TransportService transportService = Mockito.spy(extensionsRunner.createTransportService(settings));
+    public void testTransportServiceStarted() throws Exception {
 
         // verify mocked object interaction in manager method
         extensionsRunner.startTransportService(transportService);
-        Mockito.verify(transportService, times(1)).start();
+        verify(transportService, times(1)).start();
     }
 
     // test manager method invokes accept incoming requests on transport service
     @Test
-    public void testTransportServiceAcceptedIncomingRequests() throws IOException {
-
-        // retrieve and mock transport service
-        TransportService transportService = Mockito.spy(extensionsRunner.createTransportService(settings));
+    public void testTransportServiceAcceptedIncomingRequests() throws Exception {
 
         // verify mocked object interaction in manager method
         extensionsRunner.startTransportService(transportService);
-        Mockito.verify(transportService, times(1)).acceptIncomingRequests();
+        verify(transportService, times(1)).acceptIncomingRequests();
     }
+
+    @Test
+    public void testRegisterRequestHandler() throws Exception {
+
+        extensionsRunner.startTransportService(transportService);
+        verify(transportService, times(3)).registerRequestHandler(anyString(), anyString(), anyBoolean(), anyBoolean(), any(), any());
+    }
+
+    @Test
+    public void testHandlePluginsRequest() throws Exception {
```

Comment:
I don't see where `handlePluginsRequest()` throws an exception?

Reply:
We require an exception for `new TransportAddress(InetAddress.getByName("localhost"), 9876)` here and not for `handlePluginsRequest()`.